src/shared/bus-util.c

```diff
@@ -709,3 +711,33 @@ int bus_property_get_string_set(
 
         return bus_message_append_string_set(reply, *s);
 }
+
+void bus_log_caller(sd_bus_message *message, const char *method) {
+        _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
+        const char *comm = NULL, *caller = NULL;
+        _cleanup_free_ char *comm_parent = NULL;
+        pid_t pid, ppid = 0;
+
+        assert(message);
+        assert(method);
+
+        if (sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID|SD_BUS_CREDS_AUGMENT|SD_BUS_CREDS_COMM|SD_BUS_CREDS_UNIT, &creds) < 0)
+                return;
+
+        /* We need at least the PID, otherwise there's nothing to log, the rest is optional */
+        if (sd_bus_creds_get_pid(creds, &pid) < 0)
+                return;
+
+        (void) sd_bus_creds_get_comm(creds, &comm);
+        (void) sd_bus_creds_get_unit(creds, &caller);
+
+        if (!caller)
+               cg_pid_get_unit(pid, (char **) &caller);
+
+        if (get_process_ppid(pid, &ppid) < 0)
```

Comment:
As already mentioned plenty of times, drop this

Reply:
Calling `cg_pid_get_unit` is necessary since `sd_bus_creds` don't always contain the caller's unit. The unit is not there when `daemon-reload` is run, likely the same is true for `daemon-reexec`. In the original `log_caller`, the unit is retrieved from `Manager` which is something I cannot do anymore. 
.
When it comes to the rest, i.e., caller's parent's pid and command, I can remove them but it will decrease the value of the entire feature. The logged messages will look like the ones from my Sep 9 comment.